### PR TITLE
Added list item, hover and divider style

### PR DESCRIPTION
**Pull request by KhaledNjim**
## Screenshot
image
# web
image

### CODE REVIEW

File: lib/style/linagora_hover_style.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import 'package:flutter/material.dart';
+import 'package:linagora_design_flutter/colors/linagora_sys_colors.dart';
+
+class LinagoraHoverStyle {
```
Comment (hoangdat) [inline lib/style/linagora_hover_style.dart:4]: how to use it?

Comment (KhaledNjim) [reply lib/style/linagora_hover_style.dart:4]: Apply the values to an inkwell widget

**Comment by nqhhdev** [comment]
- add demo in PR

File: lib/list_item/twake_list_item.dart [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+        height: ListItemStyle.height,
+        decoration: BoxDecoration(
+          border: LinagoraDividerStyle.material().dividerDecoration,
+        ),
+        child: child);
```
Comment (nqhhdev) [inline lib/list_item/twake_list_item.dart:20]:
```suggestion
    return Container(
        height: ListItemStyle.height,
        decoration: BoxDecoration(
          border: LinagoraDividerStyle.material().dividerDecoration,
        ),
        child: child,
   );
```

File: lib/list_item/twake_list_item.dart [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  const TwakeListItem({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+        height: ListItemStyle.height,
```
Comment (hoangdat) [inline lib/list_item/twake_list_item.dart:16]: should be flexible to change it in using

File: example/lib/demos/list_item_demo.dart [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(),
+        body: ListView(
+          children: [
+            Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 8.0),
+              child: TwakeInkWell(
+                  child: TwakeListItem(
+                child: Column(
```
Comment (dab246) [inline example/lib/demos/list_item_demo.dart:31]: indent

File: lib/list_item/twake_list_item.dart [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        decoration: BoxDecoration(
+          border: LinagoraDividerStyle.material().dividerDecoration,
```
Comment (hoangdat) [inline lib/list_item/twake_list_item.dart:18]: Should not use it miscellaneous. Border should be border. In the future, when divider changes, border will not be impacted

File: lib/list_item/twake_inkwell.dart [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  final Function? onTap;
+  final Function(TapDownDetails)? onSecondaryTapDown;
+  final Function? onLongPress;
+  final Widget child;
+  final bool isSelected;
+
+  const TwakeInkWell({
+    super.key,
+    this.onTap,
+    required this.child,
```
Comment (dab246) [inline lib/list_item/twake_inkwell.dart:14]: should put `required` parameters at the top after `super`

File: lib/list_item/twake_inkwell.dart [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Material(
+      borderRadius: LinagoraHoverStyle.material().borderRadius,
+      color: widget.isSelected
+          ? LinagoraHoverStyle.material().selectedColor
+          : Colors.transparent,
+      child: InkWell(
+          splashColor: LinagoraHoverStyle.material().hoverColor,
+          onTap: () => widget.onTap,
```
Comment (dab246) [inline lib/list_item/twake_inkwell.dart:36]: Can be written more concisely.
```suggestion
          onTap:  widget.onTap,
```

File: lib/list_item/twake_inkwell.dart [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+          onSecondaryTapDown: widget.onSecondaryTapDown,
+          onLongPress: (() => widget.onLongPress),
```
Comment (dab246) [inline lib/list_item/twake_inkwell.dart:38]: idem

File: lib/style/linagora_hover_style.dart (hunk 1 above)
Comment (hoangdat) [reply lib/style/linagora_hover_style.dart:4]: can you demo it. still not see it in example

File: lib/style/linagora_hover_style.dart [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+      LinagoraHoverStyle._material();
+
+  factory LinagoraHoverStyle.material() {
+    return _materialLinagoraHoverStyle;
+  }
+
+  LinagoraHoverStyle._material()
+      : selectedColor = LinagoraSysColors.material().secondaryContainer,
+        hoverColor = LinagoraSysColors.material().surface,
+        borderRadius = BorderRadius.circular(4);
```
Comment (dab246) [inline lib/style/linagora_hover_style.dart:19]:
```suggestion
        borderRadius = const BorderRadius.all(Radius.circular(4));
```

File: lib/style/list_item_style.dart [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import 'package:flutter/material.dart';
+import 'package:linagora_design_flutter/colors/linagora_ref_colors.dart';
+
+class ListItemStyle {
+  static const double height = 80.0;
+  static final BorderRadiusGeometry borderRadius = BorderRadius.circular(4);
```
Comment (dab246) [inline lib/style/list_item_style.dart:6]: idem

**Comment by hoangdat** [comment]
- still no demo, screenshot?

File: lib/style/linagora_divider_style.dart [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+  LinagoraDividerStyle._material()
+      : color = LinagoraStateLayer(LinagoraSysColors.material().surfaceTint)
+            .opacityLayer3,
+        thickness = 1.0,
+        borderDecoration = Border(
+            bottom: BorderSide(
+          width: 1.0,
+          color: LinagoraStateLayer(LinagoraSysColors.material().surfaceTint)
+              .opacityLayer3,
+        ));
```
Comment (nqhhdev) [inline lib/style/linagora_divider_style.dart:26]:
```suggestion
  LinagoraDividerStyle._material()
      : color = LinagoraStateLayer(LinagoraSysColors.material().surfaceTint)
            .opacityLayer3,
        thickness = 1.0,
        borderDecoration = Border(
            bottom: BorderSide(
          width: 1.0,
          color: LinagoraStateLayer(LinagoraSysColors.material().surfaceTint)
              .opacityLayer3,
        ),
      );
```

File: lib/style/linagora_hover_style.dart [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+      LinagoraHoverStyle._material();
+
+  factory LinagoraHoverStyle.material() {
+    return _materialLinagoraHoverStyle;
+  }
+
+  LinagoraHoverStyle._material()
+      : selectedColor = LinagoraSysColors.material().secondaryContainer,
+        hoverColor = LinagoraSysColors.material().surface,
+        borderRadius = const BorderRadius.all(Radius.circular(4));
```
Comment (nqhhdev) [inline lib/style/linagora_hover_style.dart:19]: add commas

**Review by hoangdat — APPROVED** [review]

File: lib/style/list_item_style.dart [hunk 13]
```diff
@@ -0,0 +1,10 @@
+import 'package:flutter/material.dart';
+import 'package:linagora_design_flutter/colors/linagora_ref_colors.dart';
+
+class ListItemStyle {
```
Comment (hoangdat) [inline lib/style/list_item_style.dart:4]: please add text style item too

**Review by hoangdat — APPROVED** [review]

**Pull request merged**